packages/react/src/auto/polaris/PolarisAutoBulkActionModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { Banner, Button, ButtonGroup, Modal, Spinner, Text } from "@shopify/polaris";
+import React, { useCallback, useEffect, useMemo } from "react";
+import { useBulkAction } from "../../useBulkAction.js";
+import { humanizeCamelCase } from "../../utils.js";
+import { ModelActionDetails } from "../hooks/useModelActionsForTable.js";
+
+export const PolarisAutoBulkActionModal = (props: { model: any; modelActionDetails?: ModelActionDetails; ids: string[] }) => {
+  const { model, modelActionDetails, ids } = props;
+
+  const [showModal, setShowModal] = React.useState(!!modelActionDetails);
+  const [actionName, setActionName] = React.useState(modelActionDetails?.apiIdentifier);
+
+  useEffect(() => {
+    if (!modelActionDetails) {
+      return;
+    }
+    setShowModal(true);
+    setActionName(modelActionDetails.apiIdentifier);
+  }, [modelActionDetails]);
+
+  const actionIsLoaded = modelActionDetails && actionName && model[actionName];
+  const isBulkAction = actionIsLoaded && modelActionDetails.isBulk;
+
+  const modalTitle = useMemo(() => humanizeCamelCase(actionName ?? ""), [actionName]);
+
+  if (!actionIsLoaded) {
+    return null;
+  }
+
+  return (
+    <>
+      <Modal onClose={() => setShowModal(false)} title={modalTitle} open={showModal}>
+        {
+          isBulkAction ? (
+            <BulkActionModalContent model={model} modelActionDetails={modelActionDetails} ids={ids} close={() => setShowModal(false)} />
+          ) : null
+          // TODO - Decide on support for non-bulk actions
+          // <PolarisAutoForm action={model[actionName]} />
```

Comment:
I don't think we should include stuff like this in the codes. We should probably add tickets to them instead.

Reply:
Removed. This is debatably not needed with the custom cell renderer and the new introduction of the Action button 😎